crates/fuel-core/src/coins_query.rs

```diff
@@ -709,6 +720,70 @@ mod tests {
             let (owner, asset_ids, base_asset_id, db) = setup_coins_and_messages();
             multiple_assets_helper(owner, &asset_ids, &base_asset_id, db).await;
         }
+
+        mod allow_partial {
```

Comment:
It also would be nice to add tests for the case when indexation is available

Reply:
I added the tests